File: Scala/167. Two Sum II - Input Array Is Sorted.scala
object Solution {
    def twoSum(numbers: Array[Int], target: Int): Array[Int] = {
        var a = numbers.size - 1
        var b = 0
        while(numbers(a) + numbers(b) != target){
            if(numbers(a) + numbers(b) > target){ a -= 1 }
            else{ b += 1 }
        }
        Array(b + 1, a + 1)
    }
}

File: C#/167. Two Sum II - Input Array Is Sorted.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        var a = numbers.Length - 1;
        var b = 0;
        while(numbers[a] + numbers[b] != target){
            if(numbers[a] + numbers[b] > target){ a--; }
            else{ b++; }
        }
        int[] result = {b + 1, a + 1};
        return result;
    }
}
